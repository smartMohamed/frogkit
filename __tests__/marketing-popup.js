import {
  mount,
  shallow
} from 'vue-test-utils'

import MarketingPopup from '@/components/marketing-popup'

const propsData = {
  visible: true,
  config: {
    title: 'Sorry!',
    image: true,
    headline: 'Due to high demand, our expected delivery time is currently 8 - 12 business days.',
    label: 'Voucher:',
    code: 'Sorry19',
    accept: 'Continue',
    reject: 'Later'
  }

}

describe('Coupon Popup component', () => {
  it('should render component and match snapshot', () => {
    const wrapper = mount(MarketingPopup, {
      propsData
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('Should emit change event when close method is called', () => {
    const cmp = shallow(MarketingPopup, {
      propsData
    })
    const stub = jest.fn()
    cmp.vm.$on('change', stub)
    cmp.vm.close()
    expect(stub).toBeCalled()
  })

  it('Should emit proceed event when proceed method is called', () => {
    const cmp = shallow(MarketingPopup, {
      propsData
    })
    const stub = jest.fn()
    cmp.vm.$on('proceed', stub)
    cmp.vm.proceed()
    expect(stub).toBeCalled()
  })
})