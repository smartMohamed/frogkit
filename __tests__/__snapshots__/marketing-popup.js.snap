// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Coupon Popup component should render component and match snapshot 1`] = `
<flex justify="center" align="center" class="fk-dialogue v-enter fk-dialogue__animation-in">
    <section class="fk-dialogue__frame">
        <a class="fk-dialogue__close">
            <icon icon="close-light" color="orange"></icon>
        </a>
        <div class="fk-marketing-popup">
            <title color="primary" size="xl" class="fk-marketing-popup__title">Sorry!</title> <img src="../assets/geschenk.svg" class="fk-marketing-popup__img">
            <p class="fk-marketing-popup__txt">Due to high demand, our expected delivery time is currently 8 - 12 business days.</p>
            <p class="fk-marketing-popup__txt">Voucher: <span class="fk-marketing-popup__txt__code">Sorry19</span></p> <button size="big" block="true" color="secondary" link="" class="fk-marketing-popup__btn"><span>Continue</span> <icon icon="arrow-right"></icon></button> <span class="fk-marketing-popup__close">Later</span></div>
    </section>
</flex>
`;
